### PR TITLE
Bump automate UI from Angular 8 -> Angular 9

### PR DESCRIPTION
**Pull request by msorens**
### :nut_and_bolt: Description: What code changed, and why?

This PR bumps our Angular framework from v8 to v9, current at the time of writing. Furthermore, this PR also includes freshening almost every other 3rd party package used by automate-ui, with just a few exceptions noted below. 
The steps for this global version update are detailed below.

#### Migrate Angular

This did two things:

1. Update the various `@angular` packages in package.json.
2. Automatically apply migration patches to our code base.

Commands:
`ng update @angular/core @angular/cli`
`ng update @angular/material`
`ng update @nguniversal/express-engine` (because `npm outdated` now revealed this was needed)

Reference: https://update.angular.io/#8.2:9.0l2

#### Update all other npm packages to current (or constrained) version

This was done with a combination of `npm install`, `npm update`, and `npm audit fix`.
The remaining items are all explicitly constrained:

```
$ npm outdated
Package                         Current      Wanted       Latest  Location
@nguniversal/express-engine  9.0.0-rc.2  9.0.0-rc.2        9.0.2  automate-ui
@types/jasmine                   3.3.10      3.3.10        3.5.9  automate-ui
@types/node                    12.12.29    12.12.29       13.9.0  automate-ui
diff2html                        2.12.2      2.12.2        3.1.3  automate-ui
immutable                         3.8.2       3.8.2  4.0.0-rc.12  automate-ui
rrule                             2.4.1       2.4.1        2.6.4  automate-ui
tslint                           5.20.1      5.20.1        6.1.0  automate-ui
typescript                        3.7.5       3.7.5        3.8.3  automate-ui
```

Reasons for these constraints are detailed in the README.md file.

#### Patch code incompatibilities due to all of the above

The few necessary patches, included in various commits herein, have achieved these milestones:

 - [x] build (`ng build`)
 - [x] unit tests (via wallaby in VSCode)
 - [x] unit tests (`make unit` via command-line)
 - [x] e2e tests (protractor via `make e2e`)
 - [x] lint (`make lint`)
 - [x] launch application (`make serve`)
 - [x] application runs with no errors in browser
 - [x] application runs with no warnings in browser

There are some [breaking changes](https://angular.io/guide/updating-to-version-9#new-breaking-changes) in Angular 9, but it does not appear that that broke anything for us. 🤞 

### :chains: Related Resources
Last Angular Update (v6 -> v8): #1707
Update node (v10 -> v12): #2579 
Bump other UI packages: #2576
Bump other UI packages again: #3057 (while working on Angular 9 issues)

### :+1: Definition of Done

With Angular 9 / TypeScript 3.7 ecosystem, Automate-UI and all of its unit and integration tests still function properly.

### :athletic_shoe: How to Build and Test the Change

Do your usual hab studio setup notably including building `automate-ui-devproxy`. 

Then (line numbers shown for reference):
```
[1] cd components/automate-ui
[2] mv node_modules /location/to/save/your/old/node_modules
[3] npm install
[4] make serve
```

Notes:
Line [2] is only needed if you want to be able to easily resume what you were doing before you tried out this PR! Once you are done evaluating, move or delete your new node_modules and restore your copy made in line [2].
Line [3] generates the new node_modules directory for this PR.
Line [4] starts automate-ui.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [x] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/automate-ui/README.md [hunk 1]
```diff
@@ -83,13 +83,16 @@ That is a guess on my part, could not really confirm what version one should be
```
Comment (msorens) [inline components/automate-ui/README.md]: Note: The best way to get through all these files quickly is to go through the commits one by one.

File: components/automate-ui/README.md [hunk 2]
```diff
@@ -83,13 +83,16 @@ That is a guess on my part, could not really confirm what version one should be
 
 **Package diff2html: ^2.12.2**
```
Comment (msorens) [inline components/automate-ui/README.md]: This section of the readme details the constrained packages; just a couple updates here.

File: components/automate-ui/angular.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
         "build": {
           "builder": "@angular-devkit/build-angular:browser",
           "options": {
```
Comment (msorens) [inline components/automate-ui/angular.json]: Automatic changes done by the Angular migration.

File: components/automate-ui/package.json [hunk 4]
```diff
@@ -24,23 +24,23 @@
   },
   "private": true,
   "dependencies": {
```
Comment (msorens) [inline components/automate-ui/package.json]: The key changed file; most done by the Angular migration; a few done manually by me.

File: components/automate-ui/src/app/app-routing.module.ts [hunk 5]
```diff
@@ -78,7 +78,7 @@ const routes: Routes = [
         },
         {
```
Comment (msorens) [inline components/automate-ui/src/app/app-routing.module.ts]: More automatic updates.

File: components/automate-ui/src/app/components/authorized/authorized.component.spec.ts [hunk 6]
```diff
@@ -63,7 +63,7 @@ describe('AuthorizedComponent', () => {
         AuthorizedComponent
       ]
     }).compileComponents();
-    const store: Store<NgrxStateAtom> = TestBed.get(Store);
+    const store: Store<NgrxStateAtom> = TestBed.inject(Store);
```
Comment (msorens) [inline components/automate-ui/src/app/components/authorized/authorized.component.spec.ts]: Most of the files in the PR are this breaking change that had to be updated.

File: components/automate-ui/src/styles/mixins/_mat-theming.scss [hunk 7]
```diff
@@ -168,9 +168,38 @@ $backdrop-animation-timing-function: cubic-bezier(0.25, 0.8, 0.25, 1) !default;
 //
 // @param target Which kind of high contrast setting to target. Defaults to `active`, can be
 //    `white-on-black` or `black-on-white`.
-@mixin cdk-high-contrast($target: active) {
-  @media (-ms-high-contrast: $target) {
-    @content;
+@mixin cdk-high-contrast($target: active, $encapsulation: 'any') {
```
Comment (msorens) [inline components/automate-ui/src/styles/mixins/_mat-theming.scss]: Updated this file to the latest version from angular material. (This file is an excerpt from _theming.scss and the version we had failed to compile.)

Comment (msorens) [reply components/automate-ui/src/styles/mixins/_mat-theming.scss]: That was @scottopherson's good work. 😁

**Comment by msorens** [comment]
We resolved the `ng.probe is not a function` issue: that was due to having the Augury browser extension installed -- see https://github.com/angular/angular/issues/30737

File: components/automate-ui/src/styles/mixins/_mat-theming.scss (hunk 7 above)
Comment (SEAjamieD) [reply components/automate-ui/src/styles/mixins/_mat-theming.scss]: Great work @scottopherson !

**Pull request merged**